src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -0,0 +1,24 @@
+package viper.silver.plugin.standard.inline
+
+import viper.silver.ast.utility.ViperStrategy
+import viper.silver.ast.utility.rewriter.Traverse
+import viper.silver.ast.Program
+import viper.silver.plugin.{ParserPluginTemplate, SilverPlugin}
+
+class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate with InlineRewrite {
+
+  private[this] val InlinePredicateKeyword = "inline"
+
+  override def beforeVerify(input: Program): Program = {
+    val declaredMethods = input.methods
+    val methodsWithInlinedPreds = declaredMethods.map(inlinePredicates(_, input))
+    val rewrittenMethods = methodsWithInlinedPreds.map(removeUnfoldFold(_, input))
+
+    // TODO: actually rewrite program with methods with expanded predicates
+    val rewrittenProgram = ViperStrategy.Slim({
+      case program: Program =>
+        program.copy(methods = rewrittenMethods)(program.pos, program.info, program.errT)
+    }, Traverse.BottomUp).execute(input)
+    rewrittenProgram
```

Comment:
I'm not entirely sure how these `Strategy`s work; do we need to explicitly collect the nodes we want to visit by calling `recurseFunc` before `execute`? This is what the termination plugin does, and I've commented places in my comments below where they might be needed.
Also, is there a reason why this is traversing bottom-up rather than whatever the default traversal is (top-down, probably)?
And last thing, no need to bind to `rewrittenProgram` when it's just being returned (unless there's some Scala code style you have in mind that prefers this).